apollo-ios/Sources/Apollo/ApolloStore.swift

```diff
@@ -49,7 +50,7 @@ public class ApolloStore: @unchecked Sendable {
   ///   - callbackQueue: The queue to call the completion block on. Defaults to `DispatchQueue.main`.
   ///   - completion: [optional] A completion block to be called after records are merged into the cache.
   public func clearCache(callbackQueue: DispatchQueue = .main, completion: ((Result<Void, any Swift.Error>) -> Void)? = nil) {
-    queue.async(flags: .barrier) {
+    readerWriterLock.write {
```

Comment:
This should have been part of the ApolloStore PR that is stacked on this branch, but accidentally got included here.

This shouldn't matter too much, as that branch will be merged into this before merging this up the stack.